plugins/modules/ipatrust.py

```diff
@@ -0,0 +1,282 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import six
+from ansible.module_utils.ansible_freeipa_module import temp_kinit, \
+    temp_kdestroy, valid_creds, api_connect, api_command, date_format, \
+    compare_args_ipa, module_params_get, api_check_param, api_get_realm, \
+    api_command_no_name
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import AnsibleModule
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'supported_by': 'community',
+                    'status': ['preview'],
+                    }
+
+DOCUMENTATION = """
+---
+module: ipatrust
+author: Rob Verduijn (@qstars.nl)
+short_description: Manage FreeIPA Domain Trusts.
+description:
+- This command enable you to create, delete or disable FreeIpa Domain Trusts with an Microsoft AD controller.
+  realm:
+    description:
+    - Realm name
+    required: true
+  trust_type:
+    - Trust type (ad for Active Directory, default)
+    default: ad
+    required: true
+  admin:
+    - Active Directory domain administrator
+    required: false
+  password:
+    - Active Directory domain administrator's password
+    required: false
+  server:
+    - Domain controller for the Active Directory domain (optional)
+    required: false
+  trust_secret:
+    description:
+    - Shared secret for the trust
+    required: false
+  base_id:
+    description:
+    - First Posix ID of the range reserved for the trusted domain
+    required: false
+  range_size:
+    description:
+    - Size of the ID range reserved for the trusted domain
+  range_type:
+    description:
+    - Type of trusted domain ID range, one of ipa-ad-trust, ipa-ad-trust-posix
+    default: ipa-ad-trust
+    required: false
+  two_way:
+    description:
+    - Establish bi-directional trust. By default trust is inbound one-way only.
+    default: false
+    required: false
+    choices: ["true", "false"]
+  external:
+    description:
+    - Establish external trust to a domain in another forest. The trust is not transitive beyond the domain.
+    default: false
+    required: false
+    choices: ["true", "false"]
+  state:
+    description: State to ensure
+    default: present
+    required: true
+    choices: ["present", "absent"]
+
+version_added: "2.9"
+author:
+    - Rob Verduijn
+"""
+
+EXAMPLES = """
+# add ad-trust
+- ipatrust:
+    realm: example.com
+    trust_type: ad
+    realm_admin: Administrator
+    realm_passwd: Welcome2020!
+    state: present
+
+# delete ad-trust
+- ipatrust:
+    realm: example.com
+    state: absent
+"""
+
+RETURN = """
+"""
+
+
+if six.PY3:
+    unicode = str
```

Comment:
I don't think this is needed for this module.

Reply:
removed